rpc_util_test.go

```diff
@@ -170,3 +171,33 @@ func TestBackoff(t *testing.T) {
 		}
 	}
 }
+
+// bmEncode benchmarks encoding a simple Protocol Buffer message containing
+// mSize bytes.
+func bmEncode(b *testing.B, mSize int) {
+	msg := &testpb.Buffer{Body: make([]byte, mSize)}
+	encoded, _ := encode(msg, compressionNone)
+	encodedSz := int64(len(encoded))
+	b.ReportAllocs()
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		encode(msg, compressionNone)
+	}
+	b.SetBytes(encodedSz)
+}
+
+func BenchmarkEncode1(b *testing.B) {
```

Comment:
Instead of 1, 10, 100, 1000, I would like to do 1B, 1KB, 8KB, 64KB, 512KB, 1MB. This should not have any impact on your optimization but will be more useful later.